include_core/omrport.h

```diff
@@ -748,52 +748,135 @@ typedef struct J9ProcessorInfos {
 #define OMRPORT_MMAP_SYNC_ASYNC  0x100
 #define OMRPORT_MMAP_SYNC_INVALIDATE  0x200
 
-#define OMRPORT_SIG_FLAG_MAY_RETURN  0x1
-#define OMRPORT_SIG_FLAG_MAY_CONTINUE_EXECUTION  0x2
-#define OMRPORT_SIG_SMALLEST_SIGNAL_FLAG  0x4
-#define OMRPORT_SIG_FLAG_SIGSEGV  0x4
-#define OMRPORT_SIG_FLAG_SIGBUS  0x8
-#define OMRPORT_SIG_FLAG_SIGILL  0x10
-#define OMRPORT_SIG_FLAG_SIGFPE  0x20
-#define OMRPORT_SIG_FLAG_SIGTRAP  0x40
-#define OMRPORT_SIG_FLAG_SIGABEND  0x80
-#define OMRPORT_SIG_FLAG_SIGPIPE  0x100
-#define OMRPORT_SIG_FLAG_SIGALRM  0x200
+#if defined(OSX) || defined(LINUX) || defined(AIXPPC) || defined(J9ZOS390)
```

Comment:
I suggest the implementations should be all updated to use the new signal constant pattern, otherwise it will be more difficult to manage future work in this area (by having to keep in mind there are two constant patterns).

Reply:
I agree with your concerns. But, the objective of this PR is to support all POSIX signals on UNIX (POSIX-compatible) systems.

[win32|win64amd] implementations are complete; Windows doesn't support all POSIX signals; and no immediate value will be added by using the new signal constants in the [win32|win64amd] implementations.

Major ztpf ports are to be done by the ztpf team since we don't have ztpf machines for testing. There is already a backlog of unix/omrsignal.c changes which need to be moved to ztpf/omrsignal.c. So, we will have to get the ztpf team involved.

Updating the remaining [win32|win64|ztpf] implementations to use the new signal constants should be pursued as standalone tasks. I will open an issue to track this task.

Reply:
Hmm I would have expected ALL platforms to switch to the new way of specifying flags as part of this change. It feels like a bunch of things would have been easily shared if that was the case. Do non POSIX platforms have the notion of synchronous and asynchronous signals? 
As a minimum I would have thought that these flags would have been shared on all platforms:
```
#define OMRPORT_SIG_FLAG_IS_ASYNC               ((uint32_t)0x01)
#define OMRPORT_SIG_FLAG_IS_SYNC                ((uint32_t)0x02)
#define OMRPORT_SIG_FLAG_MAY_RETURN             ((uint32_t)0x04)
#define OMRPORT_SIG_FLAG_MAY_CONTINUE_EXECUTION ((uint32_t)0x08)
#define OMRPORT_SIG_FLAG_CONTROL_BITS_MASK      ((uint32_t)0x0F)
```

Reply:
> I would have expected ALL platforms to switch to the new way of specifying flags as part of this change.

It is not possible to support all platforms in this change; and it diverges from agile practices. To use the new signal constants, the `omrsig_*` functions need to be updated on `ztpf|win32|win64amd`. As per my knowledge, major `ztpf` ports are to be done by the `ztpf` team.

[win32|win64amd] implementations are complete, and no value will be added by using the new signal constants here.

> Do non POSIX platforms have the notion of synchronous and asynchronous signals?

Yes.

> As a minimum I would have thought that these flags ...

Refer to these [changes](https://github.com/eclipse/omr/pull/3921/commits/2de8032cc5f2ef56edd87ecae083d6489d5c7703); it contains the maximum code that can be shared between the old and new signal constants without breaking anything. Incremental work will update `omrsig_*` functions on `ztpf|win32|win64amd` to use the new signal constants, and then the old signal constants can be completely removed.